Updates to `PaymentRequest` stuctures to align with the latest TSC spec.

AustEcon opened the pull request:
As discussed with @jadwahab and @sirdeggen here is the PR for updating to the latest TSC spec: https://tsc.bitcoinassociation.net/standards/direct_payment_protocol/

Corresponding dual PR here: https://github.com/bitcoin-sv/dpp-proxy/pull/57

So far this only updates the `PaymentRequest` structure which is enough for now to get ElectrumSV through the first part of the data flow for development...

Still to do:

- [x] `Payment`
- [x] `PaymentACK`
- [x] `Adequate testing end-to-end against this code`

Until the `dpp-proxy` is updated to the TSC spec, ElectrumSV will be building against this feature branch to avoid technical debt of the old message structures. So if you are not a fan of this implementation, that is no problem - it still unblocks Roger and I to keep progressing the ElectrumSV integration until the dpp-proxy master branch catches up. 😃

AustEcon committed:
Updates to `PaymentRequest` stuctures to align with the latest TSC spec.

AustEcon committed:
Fix linting errors

jadwahab commented on payment_request.go:
```diff
@@ -2,45 +2,93 @@ package dpp
+}
+
+// PaymentRequest message as defined in the DPP T$C spec.
 type PaymentRequest struct {
```
also shouldn't this be renamed to `PaymentTerms`? 🤔

AustEcon replied:
Yes. You are right. I will change it

AustEcon committed:
Updates to `Payment` and `PaymentACK` message structures for TSC compliance

- Rename `PaymentRequest` -> `PaymentTerms` as `PaymentRequest` was the old BIP270 terminology for this struct
- Update the payment.go:Validate() method for the new `Payment` struct

jadwahab commented on payment_request.go:
```diff
@@ -57,5 +104,5 @@ type PaymentRequestService interface {
 
 // PaymentRequestReader will return a new payment request.
 type PaymentRequestReader interface {
```
shouldn't all places where we have PaymentRequest be PaymentTerms instead?

AustEcon replied:
Yes. I suppose it should. I have now updated all instances of `PaymentRequest` to `PaymentTerms` for go-dpp and the dpp-proxy.

This includes the websocket message types: `paymentterms.create`, `paymentterms.response` and `paymentterms.error`

For consistency.

jadwahab reviewed (changes requested):
I think we should have different folders/modules for different payment modes, what do you think? would be worth asking @theflyingcodr

AustEcon committed:
Rename all instances of `PaymentRequest` to `PaymentTerms` & refactoring to place each of the three dpp message type structs in their own folders

AustEcon commented:
> I think we should have different folders/modules for different payment modes, what do you think? would be worth asking @theflyingcodr

I have refactored the three DPP message types into a `messages` package with:
- `hybrid_payment_terms.go`
- `hybrid_payment.go`
- `hybrid_payment_ack.go`

The reason I shied away from nesting it any further is because I personally prefer when I am using a library (in the dpp-proxy) that the code completion groups and lists out all of the relevant structs when I type `messages. `

I think it's more intuitive when you see something like:
![image](https://user-images.githubusercontent.com/8123976/182124435-37927285-834c-4ad5-b65d-2f7b8a876463.png)

rt121212121 committed:
Update for secure payment terms. (#70)

sirdeggen commented:
Still discussing, sorry @AustEcon - early next week should have resolved the differences in opinion on this.

rt121212121 committed:
Update `go-dpp` for `dpp-proxy` to expect the secure by default `JSONEnvelope` wrapped `PaymentTerms` as the standard.

rt121212121 committed:
Linting fixes (presumably CI is using latest).

AustEcon committed:
Remove `Memo` and `Error` fields of the `PaymentACK` struct because these fields do not feature in the TSC specification
